### PR TITLE
Completed testing plan

### PR DESCRIPTION
**Pull request by ColinLefter**
# Summary of Changes
- Wrote a comprehensive testing plan that covers the complete testing protocol for our project now and in the future

## Topics Covered
- Software Development Process and Testing Methodology
- Software Testing Approaches
- Our Current Approaches
- Our Future Approaches
- Test Frameworks
- Test Workflow
- Specific Details

### CODE REVIEW

**Review by Hocng7 — COMMENTED** [review]
Good initial plan for what we have discussed in the past. May need to go into more detail and explanation of our testing approaches for the completed testing plan.

File: Reporting/testingPlan.md [hunk 1]
```diff
@@ -1,6 +1,27 @@
-# Testing Approaches
+# Software Development Process and Testing Methodology
+
+We will follow a Test-Driven Development (TDD) approach whereby initial test cases are written before application components are developed. As our database is central to our feature development, test cases rely on mock user data supplied to our MongoDB Atlas database on the cloud. As we ease into the development of application components, we will proceed with black-box testing as we begin to finalize components.
```
Comment (Hocng7) [inline Reporting/testingPlan.md:3]: This change is fitting with how we design our project using mostly components.

File: Reporting/testingPlan.md [hunk 2]
```diff
@@ -1,6 +1,27 @@
...
+
+# Software Testing Approaches
```
Comment (Hocng7) [inline Reporting/testingPlan.md:5]: Nicely done with the detail on our approaches to using unit-testing for the back-end methods and black-box testing for our components.

File: Reporting/testingPlan.md [hunk 3]
```diff
@@ -1,6 +1,27 @@
...
+
+Our final approach is to conduct black-box testing. Such testing targets specific components and mocks the process of receiving user input from an isolated frontend component followed by the handling of user input by REST API endpoints. As the final level of testing, this approach is particularly rigorous as the test is three-fold:
+
+1. Testing that all form data is captured in State variables 
+2. Testing that every `onSubmit` function successfully sends an HTTP request to the corresponding API endpoints that handles that request
+3. Testing that the API endpoint successfully receives the data from the HTTP request and processes it correctly with the database
+
+## Our Current Approaches
+
+Our current approaches are strictly focused on adhoc and unit testing. These are suitable for TDD as all test cases are written before components are integrated.
```
Comment (Hocng7) [inline Reporting/testingPlan.md:19]: Accurate for what we're doing so far.

File: Reporting/testingPlan.md [hunk 4]
```diff
@@ -1,6 +1,27 @@
...
+
+## Our Future Approaches
+
+Once we finalize existing components and begin introducing more components to the application, we will proceed to black-box testing whereby we test the complete pipeline of receiving and processing user input across frontend components and API endpoints.
```
Comment (Hocng7) [inline Reporting/testingPlan.md:23]: Align well with how we proceed in our testing for the project.

**Review by Hocng7 — COMMENTED** [review]
Overall, this testing plan does provide accurate descriptions and reasoning behind our testing approaches to the Accord project. Awesome work, Colin!

**Review by ThunderIW — APPROVED** [review]
I agree with the proposed test that you guy have layout

**Review by TobyNguyen710 — APPROVED** [review]
The test matches the what the requirements needed

**Pull request merged**